django/db/models/sql/query.py

```diff
@@ -1994,7 +1994,12 @@ def split_exclude(self, filter_expr, can_reuse, names_with_path):
             filter_rhs = OuterRef(filter_rhs)
         elif isinstance(filter_rhs, F):
             filter_rhs = OuterRef(filter_rhs.name)
-        query.add_filter(filter_lhs, filter_rhs)
+        negate = False
+        if filter_rhs is None:
+            negate = True
+            query.add_filter(f"{filter_lhs}__isnull", False)
```

Comment:
Stupid question time: why is this needed? Seems like it's adding a redundant filter in the where:

```
SELECT 1 AS "a"
FROM "ticket_34597_expensive_exists_translation" u1
WHERE (u1."id" IS NOT NULL
...
```